### ISSUE TITLE
Henchman should timeout if the remote machine didn't respond for long time.

### ISSUE THREAD

**Issue by baskaran-md** [apigee/henchman#82]
One of the instance in inventory was bad such that its up but responding very slowly for ssh and other remote requests.. In this case henchman was waiting indefinitely for the command to complete. It has to detect those bad instances and timeout accordingly.

**Comment by jlin21** [comment]
How about setting standard time out period for each task. And modules can override that time delay

**Comment by sudharsh** [comment]
Modules shouldn't do it. Let it be at a task level if we need such a timeout.

**Comment by jlin21** [comment]
There would be a timeout field for each task then?  Should tasks by default have a standard timeout value?

**Comment by sudharsh** [comment]
yeah, as defined by the transport. It's transport's responsibility

**Comment by jlin21** [comment]
If a task times out.  Should it error out or just mark that state as 'skipped' or 'unreachable'

**Comment by sudharsh** [comment]
Timeout should be treated as unreachable. More I think of it, more I feel
we should standardise on the states

**Comment by jlin21** [comment]
Yeah we have a few redudant states such as ok, changed, and failure, error.

**Comment by jlin21** [comment]
Actually we should probably have a state named timeout since unreachable means henchman cannot even reach the remote system

**Comment by jlin21** [comment]
https://github.com/golang/go/wiki/Timeouts
